File: 排序/冒泡排序/bubble_sort.py
def bubble_sort(nums):
    size = len(nums)
    for i in range(size - 1, 0 , -1):
        for j in range(0, i):
            if nums[j] > nums[j + 1]:
                nums[j], nums[j + 1] = nums[j + 1], nums[j]

nums = [11, 41, 5, 14, 1, 1919, 810]
bubble_sort(nums=nums)
print(nums)

File: 排序/冒泡排序/bubble_sort.js
function bubble_sort(nums) {
    let size = nums.length;
    for (let i = size - 1; i > 0; i--) {
        for (let j = 0; j < i; j++) {
            if (nums[j] > nums[j + 1]) {
                let tmp = nums[j];
                nums[j] = nums[j + 1];
                nums[j + 1] = tmp;
                // same as
                // [nums[j], nums[j + 1]] = [nums[j + 1], nums[j]]
            }
        }
    }
}
nums = [1145, 514, 1, 91, 9, 81, 0]
bubble_sort(nums)
console.log(nums)